crates/vm/levm/src/vm.rs

```diff
@@ -637,6 +648,50 @@ impl VM {
         ))
     }
 
+    fn create_post_execution(
+        &mut self,
+        initial_call_frame: &mut CallFrame,
+        report: &mut TransactionReport,
+    ) -> Result<(), VMError> {
+        if self.is_create() {
```

Comment:
Shouldn't this if be in the immediate upper layer?

Reply:
Yeah, probably it should. I didn't do it in the first place because wanted to keep transact as clean as possible. I agree though.
https://github.com/lambdaclass/ethrex/pull/1345/commits/36b8c15b2fa38948bc146d919b5c0499c7cd314f